### PR TITLE
sms/feature/integration tests

### PR DESCRIPTION
**Pull request by kalachck**

### CODE REVIEW

File: StoreManagementService/tests/PBJ.StoreManagementService.Api.IntegrationTests/ControllerTests/Abstract/BaseControllerTest.cs [hunk 1]
```diff
@@ -1,25 +1,93 @@
...
 using PBJ.StoreManagementService.Api.IntegrationTests.Configuration;
+using PBJ.StoreManagementService.Api.IntegrationTests.Constants;
 using PBJ.StoreManagementService.Api.IntegrationTests.Managers;
 using PBJ.StoreManagementService.DataAccess.Context;
+using System.Net.Http.Headers;
 
 namespace PBJ.StoreManagementService.Api.IntegrationTests.ControllerTests.Abstract
 {
+    [ExcludeFromCodeCoverage]
     public abstract class BaseControllerTest
```
Comment (Genresal) [inline StoreManagementService/tests/PBJ.StoreManagementService.Api.IntegrationTests/ControllerTests/Abstract/BaseControllerTest.cs]: Could we try to exclude all the tests folder from coverage? I mean direct inside the pipeline configuration.

File: StoreManagementService/tests/PBJ.StoreManagementService.Api.IntegrationTests/ControllerTests/Abstract/BaseControllerTest.cs [hunk 2]
```diff
@@ -1,25 +1,93 @@
...
+                return (default, response);
+            }
+
+            var a = await response.Content.ReadAsStringAsync();
+
+            var dto = JsonConvert.DeserializeObject<TDto>(await response.Content.ReadAsStringAsync());
+
+            return (dto, response);
+        }
+
```
Comment (Genresal) [inline StoreManagementService/tests/PBJ.StoreManagementService.Api.IntegrationTests/ControllerTests/Abstract/BaseControllerTest.cs:60]: I remember we discussed it, but finally there is a lot of duplicated code, could we use method (post, get, etc) as parameter and build `HttpRequestMessage` and then use `httpClient.SendAsync(request);`

**Review by Genresal — CHANGES_REQUESTED** [review]

**Review by Genresal — APPROVED** [review]

File: StoreManagementService/tests/PBJ.StoreManagementService.Api.IntegrationTests/ControllerTests/PostControllerTests.cs [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+            //Act
+            var response = await ExecuteWithStatusCodeAsync(
+                $"{TestingConstants.PostApi}?id={0}", HttpMethod.Put, requestBody);
+
+            //Assert
+            response.StatusCode.Should().Be(HttpStatusCode.NotFound);
+        }
+
+        [Theory]
+        [InlineData(1)]
```
Comment (Glipchik) [inline StoreManagementService/tests/PBJ.StoreManagementService.Api.IntegrationTests/ControllerTests/PostControllerTests.cs]: There is no need in using this attribute for only one data case. Either add more cases, or move this to constant in the case or outside of it

File: .github/workflows/sonarcloud.yml [hunk 4]
```diff
@@ -18,34 +18,40 @@ jobs:
...
         run: dotnet tool install --global JetBrains.dotCover.GlobalTool
+        
       - name: Build and analyze
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
           EXCLUDE_FROM_COVERAGE: "**/Program.cs, **/Middlewares/*, **/Validators/*, **/Extensions/*,
-            **/Exceptions/*, **/Constants/*, **/Mappers/*, **/Context/**/, **/Migrations/*"
+            **/Exceptions/*, **/Constants/*, **/Mappers/*, **/Context/**/, **/Migrations/*,
+            **/Configuration/*, **/FixtureCustomizations/*, **/Managers/*, **/ControllerTests/**/"
```
Comment (Glipchik) [inline .github/workflows/sonarcloud.yml]: Why controller tests are excluded from coverage? :)

**Review by Glipchik — CHANGES_REQUESTED** [review]

**Review by Glipchik — APPROVED** [review]

**Pull request merged**